libfreerdp/core/transport.c

```diff
@@ -773,6 +838,82 @@ SSIZE_T transport_parse_pdu(rdpTransport* transport, wStream* s, BOOL* incomplet
 			}
 		}
 	}
+	else if (transport->RdstlsMode)
+	{
+		UINT16 version = (header[1] << 8) | header[0];
```

Comment:
Might be a good idea to use the `Stream_Read_UINT16` et al functions here. old code uses this style, but for new code it might be the better choice.